uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,453 @@ define([
+     *
+     * @param {boolean} filterByStatus - if true then leave request api will be filtered using
+     * selected leave request status in the left navigation bar, which would be used to show the
+     * numbers of different statuses
+     * @return {Object}
+     */
+    function leaveRequestFilters(filterByStatus) {
+      var filters = vm.filters.leaveRequest;
+
+      return {
+        managed_by: vm.contactId,
+        type_id: filters.selectedAbsenceTypes ? filters.selectedAbsenceTypes.id : null,
+        status_id: prepareStatusFilter(filterByStatus),
+        from_date: {
+          from: filters.selectedPeriod.start_date
+        },
+        to_date: {
+          to: filters.selectedPeriod.end_date
+        },
+        contact_id: prepareContactID()
+      };
+    }
+
+    /**
+     * Loads the regions option values
+     *
+     * @return {Promise}
+     */
+    function loadRegions() {
+      return OptionGroup.valuesOf('hrjc_region')
+        .then(function (regions) {
+          vm.regions = regions;
+        });
+    }
+
+    /**
+     * Loads the status option values
+     *
+     * @return {Promise}
+     */
+    function loadStatuses() {
+      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_status')
+        .then(function (statuses) {
+          vm.leaveRequestStatuses = statuses.concat(vm.leaveRequestStatuses);
+        });
+    }
+
+    /**
+     * Returns the contact ID to be used for leave request api
+     *
+     * @return {Object}
+     */
+    function prepareContactID() {
+      if (vm.filters.leaveRequest.selectedUsers) {
+        return vm.filters.leaveRequest.selectedUsers.contact_id;
+      }
+
+      return {
+        "IN": vm.filteredUsers.length ? vm.filteredUsers.map(function (data) {
+            return data.contact_id;
+          }) : ["user_contact_id"]
+      };
+    }
+
+    /**
+     * Returns the status filter to be used for leave request api
+     *
+     * @param {boolean} filterByStatus - if true then leave request api will be filtered using
+     * selected leave request status in the left navigation bar, which would be used to show the
+     * numbers of different status's
+     * @return {Object}
+     */
+    function prepareStatusFilter(filterByStatus) {
+      var filters = vm.filters.leaveRequest,
+        statusFilter = [],
+        //get the value for the waiting_approval status
+        waitingApprovalID = vm.leaveRequestStatuses.find(function (data) {
+          return data.name === 'waiting_approval';
+        }).value;
+
+      //if filterByStatus is true then add the leaveStatus to be used in the leave request api
+      if (filterByStatus && filters.leaveStatus && filters.leaveStatus.value) {
+        statusFilter.push(filters.leaveStatus.value);
+      }
+
+      //if pending_requests is true then add the waiting_approval to be used in the leave request api
+      if (filters.pending_requests && waitingApprovalID) {
+        statusFilter.push(waitingApprovalID);
```

Comment:
We also need to check for the value in the if condition, that is why I moved to a variable
`if (filters.pending_requests && waitingApprovalID)`